draft-ietf-core-dns-over-coap.md

```diff
@@ -269,6 +266,9 @@ This document defines a CoAP Content-Format number for the Internet
 media type "application/dns-message" to be the mnemonic 553 — based on the port assignment of DNS.
 This media type is defined as in {{-doh}} Section 6, i.e., a single DNS message encoded in the DNS on-the-wire format {{-dns}}.
 Both DoC client and DoC server MUST be able to parse contents in the "application/dns-message" format.
+For the purposes of this document, only OPCODE 0 (Query) is supported for DNS messages.
+Future work might provide specitications and considerations for other OPCODE.
+A DoC server that receives a query with an unknown OPCODE SHOULD indicate in its response that it does not implement that opcode bye setting the response RCODE to 4, NotImp {{-dns}} (see also {{sec:resp-examples}}).
```

Comment:
What is the exception to this SHOULD?

(I.e., why is that not a MUST?)

also:

```suggestion
A DoC server that receives a query with an OPCODE it does not implement SHOULD indicate this in its response by setting the response RCODE to 4, NotImp {{-dns}} (see also {{sec:resp-examples}}).
```

Reply:
> What is the exception to this SHOULD?
> 
> (I.e., why is that not a MUST?)

I don't think there is a normative binding to set the RCODE to a specific error by DNS. At least I cannot find that. Also, there could be other RCODEs possible, depending on the server implementation, e.g. ServFail (2), (e.g. if there are other errors in the message or elsewhere that make the server fail), NotAuth (9), if the DoC client is not authorized to communicate with the DoC server in any way.

Reply:
OK, makes sense.
So let's get rid of 2119 language and state this as a fact, and qualify it.
See updated suggestion above.